pkg/apimachinery/registered/registered.go

```diff
@@ -114,22 +114,24 @@ func IsEnabledVersion(v unversioned.GroupVersion) bool {
 	return found
 }
 
-// EnabledVersions returns all enabled versions.
-func EnabledVersions() (ret []unversioned.GroupVersion) {
-	for v := range enabledVersions {
-		ret = append(ret, v)
+// EnabledVersions returns all enabled versions.  Groups are randomly ordered, but versions within groups
+// are priority order from best to worst
+func EnabledVersions() []unversioned.GroupVersion {
+	ret := []unversioned.GroupVersion{}
+	for _, groupMeta := range groupMetaMap {
+		ret = append(ret, groupMeta.GroupVersions...)
```

Comment:
> Hrm, ok, so this is a behavior change for ordering, I think after discussion we said this might break rest mapper because it relies on extensions/v1beta1.pods to come after v1.pods

That discussion was related to a different pull where I explicitly called out the change in the description: https://github.com/kubernetes/kubernetes/pull/20829  This has never had an ordering guarantee before.